src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/ClonableRepoFactory.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+public interface IClonableRepoFactory
+{
+    public IClonableGitRepo GetRemote(string repoUrl, ILogger logger);
+}
+
+public class ClonableRepoFactory : IClonableRepoFactory
+{
+    private readonly VmrRemoteConfiguration _vmrRemoteConfig;
+
+    public ClonableRepoFactory(VmrRemoteConfiguration vmrRemoteConfig)
+    {
+        _vmrRemoteConfig = vmrRemoteConfig;
+    }
+
+    public IClonableGitRepo GetRemote(string repoUrl, ILogger logger)
+    {
+        Uri repoUri = new(repoUrl);
+
+        IClonableGitRepo gitClient;
```

Comment:
You can simplify this method by returning the `new()`-ed object directly.

You can then drop the `else`'s too